### ISSUE TITLE
Cluster reporting under replicated ranges after removing and adding nodes.

### ISSUE THREAD

**Issue by roncrdb** [cockroachdb/cockroach#41239]
**Describe the Problem**
Enterprise client(s) have run into this and I was able to reproduce right away. 

After quitting two nodes (one at a time) and adding two nodes to replace, cluster is left with 19 underreplicated ranges (1-20)

**Steps to Reproduce**

Started a 3 node local cluster: 

```
cockroach start --insecure \
--store=node1 \
--listen-addr=localhost:26257 \
--http-addr=localhost:8080 \
--join=localhost:26257,localhost:26258,localhost:26259 \
--background

cockroach start \
--insecure \
--store=node2 \
--listen-addr=localhost:26258 \
--http-addr=localhost:8081 \
--join=localhost:26257,localhost:26258,localhost:26259 \
--background

cockroach start \
--insecure \
--store=node3 \
--listen-addr=localhost:26259 \
--http-addr=localhost:8082 \
--join=localhost:26257,localhost:26258,localhost:26259 \
--background
```

Killed node 3 with:

`cockroach quit --host=localhost:26259 --insecure`

Node is marked as suspect in the admin UI and brought up a new node with:

```
cockroach start \
--insecure \
--store=node4 \
--listen-addr=localhost:26260 \
--http-addr=localhost:8083 \
--join=localhost:26257,localhost:26258,localhost:26259 \
--background
```

Waited for everything to rebalance once Node 3 was marked as dead, rebalancing occurred as expected without any issues. 

Proceeded to kill node 4 with: 

cockroach quit --host=localhost:26260 --insecure

Node is marked as suspect in the admin UI and I brought up a new node with: 

```
cockroach start \
--insecure \
--store=node5 \
--listen-addr=localhost:26261 \
--http-addr=localhost:8084 \
--join=localhost:26257,localhost:26258 \
--background
```

Node 4 is marked as dead, but now I'm left with 19 under-replicated ranges. 

Range report shows that they are range 1-20. 

Cluster settings and zone configs are all defaults. 

Cockroach v19.1.4

Jira issue: CRDB-5473

**Comment by RoachietheSupportRoach** [comment]
Zendesk ticket #[3681](https://cockroachdb.zendesk.com/agent/tickets/3681) has been linked to this issue.

**Comment by knz** [comment]
fixed by #41328.
